src/CloudSecrets/SecretsAnalyzer.cs

```diff
@@ -137,8 +137,9 @@ public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
 
         private bool IsRuleActive(string ruleKey)
         {
+            var rulesSettings = ruleSettingsProvider.Get();
```

Comment:
The reason I Get() the rule settings everytime we check if a rule is active is so that if we update the Quality profile at any point during runtime it gets refreshed here aswell, else we'd be stuck with the same one.